meta-intel-edison-bsp/recipes-kernel/bcm43340/bcm43340-fw.bb

```diff
@@ -21,15 +23,15 @@ LIC_FILES_CHKSUM = "file://edison-firmware/broadcom_cws/wlan/firmware/LICENCE.br
 PV = "6.20.190"
 PR = "r2"
 
-FILES:${PN} += "/lib/firmware/brcm/*"
+FILES:${PN} += "${libdir}/firmware/brcm/*"
 
 inherit allarch update-alternatives
 
 do_install() {
-        install -v -d  ${D}/lib/firmware/brcm/
-        install -m 0755 edison-firmware/broadcom_cws/wlan/firmware/bcmdhd_aob.cal_4334x_b0 ${D}/lib/firmware/brcm/brcmfmac43340-sdio-fr.txt
-        install -m 0755 edison-firmware/broadcom_cws/wlan/firmware/bcmdhd.cal_4334x_b0 '${D}/lib/firmware/brcm/brcmfmac43340-sdio.Intel Corporation-Merrifield.txt'
-        install -m 0755 linux-firmware/cypress/cyfmac43340-sdio.bin ${D}/lib/firmware/brcm/brcmfmac43340-sdio.bin
```

Comment:
The Cypress FW are available in linux-firmware repository, I'm wondering why we don't use them directly.

Reply:
That is what we do here, line #14 points to the repo which causes that to be cloned and checked out.
Then we override the do_install() to install cyfmac43340-sdio.bin  to the install dir. We rename it brcmfmac43340-sdio.Intel Corporation-Merrifield.bin what the kernel seems to be looking for (is that wrong?).
From there it gets packaged automatically.